Automattic/newspack-blocks
# Add a script to generate translations

### All Submissions:


### Changes proposed in this Pull Request:

This standardizes the process to generate all files needed for a translated Gutenberg UI, improving on #307.

In #307 not all necessary JSON files were generated because the POT file didn't include references to the webpack generated files. This introduces a `npm run i18n` which
1. compiles the blocks to production ready versions, then
2. extracts the strings using `wp i18n make-pot`, then
3. updates the translations from previous PO files to maintain them (using `msgmerge`), and finally
4. generates the JSON files using `wp i18n make-json`.

### How to test the changes in this Pull Request:

1. Switch your wp-admin UI to Spanish or German
2. Add a new Homepage Posts block
3. Observe translations in the sidebar, for example "Authors", this should be translated.

### Other information:
